### PR TITLE
Feature/social media sign up

### PR DESCRIPTION
**Pull request by axelburling**

### CODE REVIEW

File: frontend/src/views/Login/Login.svelte [hunk 1]
```diff
@@ -1,14 +1,53 @@
...
+  export let onGoogleLogin: () => Promise<void>;
+  export let onGithubLogin: () => Promise<void>;
+    let email = "";
+    let password = "";
+    
+  
+
+
+
+// const signOutWithGoogle = async () => {
```
Comment (albinr99salt) [inline frontend/src/views/Login/Login.svelte:17]: this looks a bit half done, and shoudl be cleaned up

File: server/src/entity/User.ts [hunk 2]
```diff
@@ -18,9 +18,22 @@ export class User extends BaseEntity {
   @Column({ name: 'user_email', type: 'varchar', length: '255' })
```
Comment (albinr99salt) [inline server/src/entity/User.ts:18]: we should set email nullable false. email should always be required.

File: server/src/entity/User.ts [hunk 3]
```diff
@@ -18,9 +18,22 @@ export class User extends BaseEntity {
...
   email: string;
 
-  @Column({ name: 'user_password', type: 'varchar', length: '255' })
+  @Column({
+    name: 'user_password',
+    type: 'varchar',
+    length: '255',
+    nullable: true,
```
Comment (albinr99salt) [inline server/src/entity/User.ts:25]: nullable should be false, password should be required.

File: frontend/src/App.svelte [hunk 4]
```diff
@@ -1,86 +1,128 @@
...
+  };
 
+  console.log(screen);
 </script>
 
 <main>
   <Navbar />
   {#if screen === "Register"}
-    <Register setScreen={setScreen} onRegister={register}/>
+    <Register {setScreen} onRegister={register} />
```
[the remark this answers is not in the document]
Comment (axelburling) [reply frontend/src/App.svelte:124]: Ask github copilot

**Review by RAP2G — APPROVED** [review]

**Pull request merged**